File: 2.iOS界面开发（基础篇）/4.UINavigationController/4.单元作业/作业批改/Navigation/Navigation/NextViewController.m
//
//  NextViewController.m
//  Navigation
//
//  Created by Willei Wang on 2017/6/3.
//  Copyright © 2017年 Willei Wang. All rights reserved.
//

#import "NextViewController.h"

@interface NextViewController ()

@end

@implementation NextViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view.
    
    self.title = @"标题";
    
    UIBarButtonItem *rightItem = [[UIBarButtonItem alloc] initWithImage:[UIImage imageNamed:@"share"] style:UIBarButtonItemStylePlain target:self action:@selector(shareTip)];
    self.navigationItem.rightBarButtonItem = rightItem;
    
        UIBarButtonItem *leftBackItem = [[UIBarButtonItem alloc] initWithImage:[UIImage imageNamed:@"back"] style:UIBarButtonItemStylePlain target:self action:@selector(goBack)];

    
    UIBarButtonItem *closeItem = [[UIBarButtonItem alloc] initWithImage:[UIImage imageNamed:@"close"] style:UIBarButtonItemStylePlain target:self action:@selector(closeTip)];
    
    self.navigationItem.leftBarButtonItems = @[leftBackItem, closeItem];
    
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

- (void)goBack
{
    [self.navigationController popViewControllerAnimated:YES];
}

- (void)closeTip
{
    NSLog(@"关闭");
}

- (void)shareTip
{
    NSLog(@"分享");
}
/*
#pragma mark - Navigation

// In a storyboard-based application, you will often want to do a little preparation before navigation
- (void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender {
    // Get the new view controller using [segue destinationViewController].
    // Pass the selected object to the new view controller.
}
*/

@end

File: 2.iOS界面开发（基础篇）/4.UINavigationController/4.单元作业/作业批改/NavigationBar/NavigationBar/NextViewController.swift
//
//  NextViewController.swift
//  NavigationBar
//
//  Created by 胡诚真 on 2017/5/4.
//  Copyright © 2017年 胡诚真. All rights reserved.
//

import UIKit

class NextViewController: UIViewController {

    override func viewDidLoad() {
        super.viewDidLoad()

        // Do any additional setup after loading the view.
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }
    
    @IBAction func tapBackButton(_ sender: Any) {
        self.navigationController?.popViewController(animated: true)
    }

    @IBAction func share(_ sender: Any) {
        print("share")
    }
    @IBAction func close(_ sender: Any) {
        print("close")
    }
    /*
    // MARK: - Navigation

    // In a storyboard-based application, you will often want to do a little preparation before navigation
    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
        // Get the new view controller using segue.destinationViewController.
        // Pass the selected object to the new view controller.
    }
    */

}
